app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -651,6 +656,233 @@ class BrowserRobot {
         assertTrue(hintMessage.waitForExists(waitingTime))
     }
 
+    fun clickForm(
+        formType: String,
+        calendarForm: Boolean = false,
+        clockForm: Boolean = false,
+        colorPicker: Boolean = false,
+    ) {
+        mDevice.waitForObjects(webPageItemContainingText(formType))
+
+        if (calendarForm) {
+            clickPageObject(webPageItemWithResourceId("calendar"))
+            mDevice.waitForIdle(waitingTime)
+        } else if (clockForm) {
+            clickPageObject(webPageItemWithResourceId("clock"))
+            mDevice.waitForIdle(waitingTime)
+        } else if (colorPicker) {
+            clickPageObject(webPageItemWithResourceId("colorPicker"))
+            mDevice.waitForIdle(waitingTime)
+        } else {
+            clickPageObject(webPageItemWithResourceId("dropDown"))
+            mDevice.waitForIdle(waitingTime)
```

Comment:
these multiple if/else statements can be replaced with when expressions

Reply:
Done !